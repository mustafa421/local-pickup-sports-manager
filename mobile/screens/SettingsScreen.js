import React, { Component } from "react";
import { View, Text, Platform, Dimensions, StyleSheet } from "react-native";
import { Button } from "react-native-elements";
import { NavigationActions } from "react-navigation";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { doLogout } from "../actions/auth_actions";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-evenly",
    backgroundColor: "paleturquoise",
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").height
  },
  buttons: {
    margin: 5,
    padding: 10,
    flex: 1
  }
});

export class SettingsScreen extends Component {
  // A feature from React Navigation to define a title
  static navigationOptions = ({ navigation }) => ({
    title: "Profile",
    headerRight: (
      <Button
        title="Edit"
        textStyle={{ color: "rgba(0, 122, 255, 1)" }}
        onPress={() => navigation.navigate("updateSettingsScreen")}
        backgroundColor="paleturquoise"
        buttonStyle={{ borderColor: "black", borderWidth: 1, borderRadius: 5 }}
      />
    ),
    headerStyle: {
      backgroundColor: "#03A9F4",
      marginTop: Platform.OS === "android" ? 24 : 0 // To prevent overlapping from header in Android devices
    }
  });

  render() {
    const { username, email, navigation, dispatch } = this.props;
    return (
      <View style={styles.container}>
        <Text
          style={{
            padding: 50,
            color: "blue"
          }}
        >
          Name:
          {username}
        </Text>
        <Text
          style={{
            padding: 50,
            color: "blue"
          }}
        >
          Email:
          {email}
        </Text>

        <Button
          raised
          containerStyle={styles.buttons}
          onPress={() => navigation.navigate("PreferencesScreen")}
          title="Game Preferences"
          backgroundColor="#0000FF"
        />
        <Button
          raised
          title="Logout"
          backgroundColor="#E31414"
          containerStyle={styles.buttons}
          onPress={() => {
            dispatch(doLogout(navigation));
            // Reset history so we won't directed back to the settings screen
            navigation.reset(
              [NavigationActions.navigate({ routeName: "mainScreen" })],
              0
            );
            navigation.navigate("welcome");
          }}
        />
      </View>
    );
  }
}

const mapStateToProps = state => state.auth.userAccountData;

export default connect(mapStateToProps)(SettingsScreen);

SettingsScreen.propTypes = {
  dispatch: PropTypes.func,
  username: PropTypes.string,
  email: PropTypes.string
};

SettingsScreen.defaultProps = {
  dispatch: null,
  username: "",
  email: ""
};
