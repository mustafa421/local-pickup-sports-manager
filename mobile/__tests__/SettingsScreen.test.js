import React from "react";
import renderer from "react-test-renderer";
import SettingsScreen from "../screens/SettingsScreen";

describe("Settings screen test", () => {
  test("snapshot", () => {
    const tree = renderer.create(<SettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
