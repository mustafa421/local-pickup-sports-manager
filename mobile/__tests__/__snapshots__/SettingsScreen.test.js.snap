// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings screen test snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "blue",
        "padding": 50,
      }
    }
  >
    Name:
    
  </Text>
  <Text
    style={
      Object {
        "color": "blue",
        "padding": 50,
      }
    }
  >
    Email:
    
  </Text>
  <Text
    style={
      Object {
        "color": "blue",
        "padding": 50,
      }
    }
  >
    Phone Number:
    Not provided
  </Text>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#E31414",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
